pkg/platform/model/buildplanner/build.go

```diff
@@ -74,12 +74,12 @@ func (b *BuildPlanner) FetchCommit(commitID strfmt.UUID, owner, project string,
 		return nil, errs.Wrap(err, "failed to unmarshal build plan")
 	}
 
-	expression, err := buildexpression.New(commit.Expression)
+	script, err := buildscript.UnmarshalBuildExpression(commit.Expression, ptr.To(time.Time(commit.AtTime)))
```

Comment:
At this point I question the `UnmarshalBuildExpression()` name because if I'm looking to create a buildscript from an incoming buildexpression (and atTime), the last thing that comes to mind is this name, and it's not something I'd see in an autocompletion list and go "oh, that's what I need". I prefer `NewFromBuildExpression()` or something like that. That's just me though. Maybe this is a Go convention or something similar.

Reply:
I'd like to stick with the Marshal/Unmarshal terms, as that's been a pattern throughout these refactors. If this becomes a maintenance overhead we can revisit.